frontend/src/components/@common/Map/OverlayMarker.tsx

```diff
@@ -1,37 +1,82 @@
-import styled from 'styled-components';
+import styled, { keyframes } from 'styled-components';
+import { useRef, useState } from 'react';
 import ProfileImage from '../ProfileImage';
 import Overlay from './Overlay/Overlay';
 import type { Celeb } from '~/@types/celeb.types';
-import type { Coordinate } from '~/@types/map.types';
+import { Restaurant } from '~/@types/restaurant.types';
+import RestaurantCard from '~/components/RestaurantCard';
+import type { Quadrant } from '~/utils/getQuadrant';
+import useOnClickOutside from '~/hooks/useOnClickOuside';
 
 interface OverlayMarkerProps {
   celeb: Celeb;
-  position: Coordinate;
-  onClick: ({ lat, lng }: Coordinate) => void;
   map?: google.maps.Map;
+  restaurant: Restaurant;
+  quadrant: Quadrant;
 }
 
-function OverlayMarker({ celeb, position, map, onClick }: OverlayMarkerProps) {
+function OverlayMarker({ celeb, restaurant, map, quadrant }: OverlayMarkerProps) {
+  const { lat, lng } = restaurant;
+  const [isClicked, setIsClicked] = useState(false);
+  const ref = useRef();
+  useOnClickOutside(ref, () => setIsClicked(false));
+
   return (
     map && (
-      <Overlay position={position} map={map}>
-        <StyledMarker type="button" onClick={() => onClick(position)}>
-          <ProfileImage name={celeb.name} imageUrl={celeb.profileImageUrl} size={32} border />
+      <Overlay position={{ lat, lng }} map={map} zIndex={isClicked ? 18 : 0}>
+        <StyledMarker onClick={() => setIsClicked(true)} isClicked={isClicked} ref={ref}>
+          <ProfileImage name={celeb.name} imageUrl={celeb.profileImageUrl} border />
         </StyledMarker>
+        {isClicked && (
+          <StyledModal quadrant={quadrant}>
+            <RestaurantCard restaurant={restaurant} onClick={() => {}} type="map" />
```

Comment:
onClick 은 지워주어도 좋을 것 같아요!
```suggestion
            <RestaurantCard restaurant={restaurant} type="map" />
```